cdc/tests/integration_tests/sigstop/run.sh

```diff
@@ -0,0 +1,119 @@
+#!/bin/bash
+
+set -eu
+
+CUR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+source $CUR/../_utils/test_prepare
+WORK_DIR=$OUT_DIR/$TEST_NAME
+CDC_BINARY=tikv-cdc.test
+SINK_TYPE=$1
+UP_PD=http://$UP_PD_HOST_1:$UP_PD_PORT_1
+DOWN_PD=http://$DOWN_PD_HOST:$DOWN_PD_PORT
+
+function run_kill_upstream() {
+	rm -rf $WORK_DIR && mkdir -p $WORK_DIR
+	start_tidb_cluster --workdir $WORK_DIR --multiple-upstream-pd "true"
+	cd $WORK_DIR
+
+	start_ts=$(get_start_ts $UP_PD)
+	run_cdc_server --workdir $WORK_DIR --binary $CDC_BINARY --logsuffix "1" --addr "127.0.0.1:8600" --pd "$UP_PD"
+	run_cdc_server --workdir $WORK_DIR --binary $CDC_BINARY --logsuffix "2" --addr "127.0.0.1:8601" --pd "$UP_PD"
+	run_cdc_server --workdir $WORK_DIR --binary $CDC_BINARY --logsuffix "3" --addr "127.0.0.1:8602" --pd "$UP_PD"
+
+	case $SINK_TYPE in
+	tikv) SINK_URI="tikv://${DOWN_PD_HOST}:${DOWN_PD_PORT}" ;;
+	*) SINK_URI="" ;;
+	esac
+
+	tikv-cdc cli changefeed create --start-ts=$start_ts --sink-uri="$SINK_URI"
+
+	# send sigstop to tikv
+	n=$(echo $(($RANDOM % 3 + 1)))
```

Comment:
The workload would be better to cover the whold process of chaos, to verify that chaos would not affect the correctness.

E.g. a big single workload:

```bash
	rawkv_op $UP_PD put 100000 # should be BIG enough to complete after `check_count 3 "tikv-cdc" $UP_PD`.
	sleep 1

	# send sigstop to tikv
	n=$(echo $(($RANDOM % 3 + 1)))
	tikv_pid=$(pgrep -f "tikv$n" | head -n1)
	kill -19 $tikv_pid
	sleep 10
	check_count 2 "tikv" $UP_PD

	kill -18 $tikv_pid
	check_count 3 "tikv" $UP_PD

 	check_sync_diff $WORK_DIR $UP_PD $DOWN_PD
```

It would be ok if you would like to verify in the middle:

```bash
	rawkv_op $UP_PD put 100000 # should be BIG enough to complete after `check_count 2 "tikv-cdc" $UP_PD`.
	sleep 1

	# send sigstop to tikv
	n=$(echo $(($RANDOM % 3 + 1)))
	tikv_pid=$(pgrep -f "tikv$n" | head -n1)
	kill -19 $tikv_pid
	sleep 10
	check_count 2 "tikv" $UP_PD

 	check_sync_diff $WORK_DIR $UP_PD $DOWN_PD

	# resume
	rawkv_op $UP_PD delete 100000 # should be BIG enough to complete after `check_count 3 "tikv-cdc" $UP_PD`.
	sleep 1

	kill -18 $tikv_pid
	check_count 3 "tikv" $UP_PD

 	check_sync_diff $WORK_DIR $UP_PD $DOWN_PD
```

The same to rest.

Reply:
Good suggestion! And I use `rawkv_op $UP_PD put 100000 &` instead of  `rawkv_op $UP_PD put 100000`, so we can put data in backgroud.